### ISSUE TITLE
Straight-freeze-versions fails when some packages are marked built-in

### ISSUE THREAD

**Issue by ThibautVerron** [raxod502/straight.el#470]
## What's wrong

If there are some packages marked "built-in", `straight-freeze-versions` fails with "VC backend 'built-in' does not implement method 'get commit'".

It makes sense, but there is no reason why it's not possible to freeze the versions of the non-built-in packages?

## Directions to reproduce

(I'm very surprised at the instructions here. I thought the whole point of straight.el is that it is _easy_ to get reproducible behaviors. Moving .emacs.d around is not easy, some packages expect to be able to read or write there.)

```
(setq straight-use-package-by-default t
      straight-repository-branch "develop")

(defvar bootstrap-version)
(let ((bootstrap-file
       (expand-file-name "straight/repos/straight.el/bootstrap.el"
	       user-emacs-directory)) ;; 
      (bootstrap-version 5))
  (unless (file-exists-p bootstrap-file)
    (with-current-buffer
        (url-retrieve-synchronously
         "https://raw.githubusercontent.com/raxod502/straight.el/develop/install.el"
         'silent 'inhibit-cookies)
      (goto-char (point-max))
      (eval-print-last-sexp)))
  (load bootstrap-file nil 'nomessage))

(straight-use-package 'use-package)

(use-package emacs
  :straight (:type built-in))
```

Then `M-x straight-freeze-versions`.

### Version information

* Emacs version: 26.3
* Operating system: tested on Windows and GNU/Linux

**Comment by raxod502** [comment]
Should be fixed now, thanks for the report and let me know if you have any questions.

**Comment by raxod502** [comment]
For isolating an Emacs environment, one really has to move all of `~/.emacs.d`, and there's no real way to work around that. Personally, I have a snippet in my `init.el` that reads the environment variable `$USER_EMACS_DIRECTORY`, and if it's set to something then it sets the value of `user-emacs-directory` and instead loads the init-file found there. You can see the code [here](https://github.com/raxod502/radian/blob/66ecf92ddc1343444fb422657eece8b542dbc524/emacs/init.el#L9). Alternatively, you could use [chemacs](https://github.com/plexus/chemacs) or another similar tool.

I then have a shell alias for `USER_EMACS_DIRECTORY=$PWD e` which makes starting an isolated Emacs instance very easy.

However, perhaps `straight.el` could provide some such code out of the box so that this process can be automated for users. Would you like to open another issue suggesting as such?

**Comment by shackra** [comment]
this bug is still happening today for me

**Comment by shackra** [comment]
my branch of straight points to master, tho

**Comment by raxod502** [comment]
This thread is being closed automatically by [Tidier](https://github.com/raxod502/tidier) because it is labeled with "waiting on response" and has not seen any activity for 90 days. But don't worry—if you have any information that might advance the discussion, leave a comment and I will be happy to reopen the thread :)
